src/luascript.cpp

```diff
@@ -9159,9 +9162,37 @@ int LuaScriptInterface::luaPlayerSetBankBalance(lua_State* L)
 	return 1;
 }
 
+int LuaScriptInterface::luaPlayerHasStorage(lua_State* L)
+{
+	// player:hasStorage(key)
+	Player* player = getUserdata<Player>(L, 1);
+	if (!player) {
+		lua_pushnil(L);
+		return 1;
+	}
+
+	lua_pushboolean(L, player->storageMap.find(getNumber<uint32_t>(L, 2)) != player->storageMap.end());
```

Comment:
use the TFS API to push booleans onto the stack. `pushBoolean`

Reply:
```suggestion
	pushBoolean(L, player->storageMap.find(getNumber<uint32_t>(L, 2)) != player->storageMap.end());
```